src/client/datascience/jupyter/kernels/kernel.ts

```diff
@@ -883,13 +881,14 @@ export class Kernel implements IKernel {
                         : `matplotlib.rcParams.update(${Identifiers.MatplotLibDefaultParams})`
                 );
             }
-        } else {
-            const configInit = settings && settings.generateSVGPlots ? CodeSnippets.ConfigSvg : CodeSnippets.ConfigPng;
-            traceInfoIfCI(
-                `Initialize config for plots for ${(this.resourceUri || this.notebookDocument.uri).toString()}`
-            );
-            results.push(...configInit.splitLines({ trim: false }));
         }
+
+        // Add in SVG to the figure formats if needed
+        if (settings.generateSVGPlots) {
+            results.push(...CodeSnippets.AppendSVGFigureFormat.splitLines({ trim: false }));
+            traceInfo('Add SVG to matplotlib figure formats');
```

Comment:
I can no see how startup code will fail completely if users don't have matplotlib installed.
Importing packages too would fail as we won't be adding the site packages folder into the current path, i'm going to look into that now.

Reply:
@DonJayamanne Just want to make sure we are not looking at the same thing. I was also going to take a peek at hardening the startup code against exceptions as part of this PR (it's why I've not pushed yet). Couldn't quite tell from your comment above if you were as well.

Reply:
Didn't realize you were doing the same thing, FYI we have a seprate issue for hardening startup code here https://github.com/microsoft/vscode-jupyter/issues/8988
I'll close my PR.
I don't think we should wrap all startup code in try catch. I think something like what I've done would be better, so that we log the errors, Else we'd never know what went wrong.
I think should put a try..except only around the matplot lib code (as users might not have matplot lib).
& the rest of code should not fail except for initializing of the `__vsc_ipynb_file__ ` variable

As for user code, I think we shouldn't wrap that either. I think that should be a seprate request.
If there are failures it should be logged (i don't think we shoudl display messageboxes), and don't think we should add try...except for user code, might not be what they're expecting, if its their code, they should be responsible for that.

Reply:
I looked at that issue as I felt that if users don't have matplot lib then debugger code will not get setup, hence debugging will not work, i.e. that's a regression.
Hence picked that.

Reply:
@DonJayamanne I wasn't saying to can your PR, I just missed the mail on it and I was working on something similar. I had the same realization about the matplotlib code (seeing that it could error out and mess up the debugger / user code that comes after it) so I didn't want to go forward with this PR without that fixed. I'd rather do the right thing on this PR so it can be submitted in good shape. Since you closed your PR I'd go ahead with this but I'll look at your PR logging changes and look at integrating that into this PR as well.

I think with the try excepts it does get logged though, at least on CI the output stream from the print on the except message is show in the log.

Reply:
All good, 

> t does get logged though, at least on CI the output stream from the print on t

Awesome.

Reply:
No harm done, there are 600 more issues :) to work on.